integration_tests/src/main/python/parquet_write_test.py

```diff
@@ -544,6 +544,19 @@ def test_fallback_to_single_writer_from_concurrent_writer(spark_tmp_path, aqe_en
             {"spark.sql.adaptive.enabled": aqe_enabled},
         ))
 
+def test_write_exceed_max_file_counter_error(spark_tmp_path):
+    gen_list = [("c1", BooleanGen())]
```

Comment:
I'don't add tests for concurrent writer because it is very time-consuming.
But I've verified on my desktop with smaller `MAX_FILE_COUNTER`